src/verify_cert.rs

```diff
@@ -90,36 +91,40 @@ fn build_chain_inner(
 
     // TODO: revocation.
 
-    let result = loop_while_non_fatal_error(opts.trust_anchors, |trust_anchor: &TrustAnchor| {
-        let trust_anchor_subject = untrusted::Input::from(trust_anchor.subject);
-        if cert.issuer != trust_anchor_subject {
-            return Err(Error::UnknownIssuer);
-        }
+    let result = loop_while_non_fatal_error(
+        Error::UnknownIssuer,
+        opts.trust_anchors,
+        |trust_anchor: &TrustAnchor| {
+            let trust_anchor_subject = untrusted::Input::from(trust_anchor.subject);
+            if cert.issuer != trust_anchor_subject {
+                return Err(Error::UnknownIssuer);
+            }
 
-        let name_constraints = trust_anchor.name_constraints.map(untrusted::Input::from);
+            let name_constraints = trust_anchor.name_constraints.map(untrusted::Input::from);
 
-        untrusted::read_all_optional(name_constraints, Error::BadDer, |value| {
-            subject_name::check_name_constraints(value, cert, subject_common_name_contents)
-        })?;
+            untrusted::read_all_optional(name_constraints, Error::BadDer, |value| {
+                subject_name::check_name_constraints(value, cert, subject_common_name_contents)
+            })?;
 
-        // TODO: check_distrust(trust_anchor_subject, trust_anchor_spki)?;
+            // TODO: check_distrust(trust_anchor_subject, trust_anchor_spki)?;
 
-        check_signatures(
-            opts.supported_sig_algs,
-            cert,
-            trust_anchor,
-            &opts.revocation,
-        )?;
+            check_signatures(
+                opts.supported_sig_algs,
+                cert,
+                trust_anchor,
+                &opts.revocation,
+            )?;
 
-        Ok(())
-    });
+            Ok(())
+        },
+    );
 
     // If the error is not fatal, then keep going.
     if result.is_ok() {
         return Ok(());
     }
 
-    loop_while_non_fatal_error(opts.intermediate_certs, |cert_der| {
+    loop_while_non_fatal_error(result.unwrap_err(), opts.intermediate_certs, |cert_der| {
```

Comment:
If we need the error, we should do

```rust
let err = match result {
    Ok(()) => return Ok(()),
    Err(err) => err,
};
```

instead of `unwrap_err()` here.

Reply:
Done, thanks for catching that. I was being sloppy.